sdk/framework/openapi.go

```diff
@@ -200,6 +200,7 @@ var (
 	altFieldsGroupRe = regexp.MustCompile(`\(\?P<\w+>\w+(\|\w+)+\)`) // Match named groups that limit options, e.g. "(?<foo>a|b|c)"
 	altFieldsRe      = regexp.MustCompile(`\w+(\|\w+)+`)             // Match an options set, e.g. "a|b|c"
 	nonWordRe        = regexp.MustCompile(`[^\w]+`)                  // Match a sequence of non-word characters
+	altRootsRe       = regexp.MustCompile(`\(((\w*)[\||\)])+\/`)     // Paths that start with alts, e.g. "(creds|sts)/(?P<name>regex)"
```

Comment:
It looks like this will only match on the prefix (e.g. "(creds|sts)/"). Should that be in the example comment?

I believe this part `[\||\)]` can be simplified to `[|\)]`

The regex will also match something like `(creds)sts)/` or `(||)|))/`  (not sure if that's an issue)

Reply:
Since this only operates on `Pattern` s that are used in `Path`s I'm assuming that only valid regex patterns would land here. Your concerns are definitely valid though, I'll add some more tests and make sure this doesn't introduce something weird.